### PR TITLE
added toysgd benchmark

### PR DESCRIPTION
**Pull request by benjamc**
Please have a look. :-)
Especially on the bounds for the coefficients and the initial x (not sure about that).

### CODE REVIEW

File: dacbench/benchmarks/toysgd_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import numpy as np
+import pandas as pd
+from gym import spaces
+
+from dacbench.abstract_benchmark import AbstractBenchmark, objdict
+import importlib
+import dacbench.envs.toysgd
+importlib.reload(dacbench.envs.toysgd)
+
+HISTORY_LENGTH = 40
```
Comment (TheEimer) [inline dacbench/benchmarks/toysgd_benchmark.py:12]: Is the history length used somewhere I don't see right now?

File: dacbench/benchmarks/toysgd_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        "Current Learning Rate",
+        "Current Momentum"
+    ],
+}
+
+DEFAULTS = objdict(
+    {
+        "action_space_class": "Box",
+        "action_space_args": [-np.inf * np.ones((2,)), np.inf * np.ones((2,))],
+        "observation_space_class": "Dict",
```
Comment (TheEimer) [inline dacbench/benchmarks/toysgd_benchmark.py:30]: I'll not say you shouldn't or can't do that, but Dict spaces are impractical without the dict wrapper. If you don't care, feel free to ignore

**Comment by TheEimer** [comment]
Initial x as 0 makes sense, I think. Without haing every run it, I can't really say if the coefficient range is good or not, but it's super easy to adapt, so I think that's good. 
So thanks for the benchmarks, I'll merge as soon as you adressed my comment (second one is optional).

File: dacbench/benchmarks/toysgd_benchmark.py (hunk 1 above)
Comment (benjamc) [reply dacbench/benchmarks/toysgd_benchmark.py:12]: oops forgot to delete, not used right now

File: dacbench/benchmarks/toysgd_benchmark.py (hunk 2 above)
Comment (benjamc) [reply dacbench/benchmarks/toysgd_benchmark.py:30]: I just took inspiration from the sgd benchmark. If that's suboptimal, feel free to remodel it :) or I could also do it tomorrow

**Comment by TheEimer** [comment]
Any updates here, @benjamc ? I found another thing to fix in the meantime, the reset doesn't seem to return a state. I missed that, but it's definitely not desired behavior ;D

**Comment by TheEimer** [comment]
Also, we may want to set an upper limit for values. The lr get huge pretty fast with actions > 1

**Comment by TheEimer** [comment]
Changes from my side:
- limited actions to stay below 1
- made reset return a state
- initialized lr with 0 (might not be smart at all, but None is worse)

Things where input would be nice:
- how can we initialize the lr without giving the agent a nan in the reset state?
- is 1 a good limit for the actions?

As soon as we settled those two, we can merge

**Comment by benjamc** [comment]
- [lr] setting the learning rate to 0 as a starter should be fine. this way the agent is definitely forced to do something :D
and the agent does not set the learning rate itself but the log learning rate 
- [actions] do not limit actions because action for lr is log

**Pull request merged**